### ISSUE TITLE
Comment attachment is different with ASI

### ISSUE THREAD

**Issue by lo1tuma** [eslint/espree#137]
Given this code example with semicolon:

```js
a(); // comment
```

then the following AST is generated.
```js
{
    "type": "Program",
    "body": [
        {
            "type": "ExpressionStatement",
            "expression": {
                "type": "CallExpression",
                "callee": {
                    "type": "Identifier",
                    "name": "a"
                },
                "arguments": []
            },
            "trailingComments": [
                {
                    "type": "Line",
                    "value": " comment"
                }
            ]
        }
    ],
    "comments": [
        {
            "type": "Line",
            "value": " comment"
        }
    ]
}
```

When the semicolon is omitted, then the `trailingComment` is attached to different node.

```js
a() // comment
```

```js
{
    "type": "Program",
    "body": [
        {
            "type": "ExpressionStatement",
            "expression": {
                "type": "CallExpression",
                "callee": {
                    "type": "Identifier",
                    "name": "a"
                },
                "arguments": [],
                "trailingComments": [
                    {
                        "type": "Line",
                        "value": " comment"
                    }
                ]
            }
        }
    ],
    "comments": [
        {
            "type": "Line",
            "value": " comment"
        }
    ]
}
```

As you can see, in the first AST the `trailingComment` is attached to the `ExpressionStatement` node, in the second AST the `trailingComment` is attached to the `CallExpression` node.

**Comment by gyandeeps** [comment]
what behavior is expected here? case 1 or case 2.
fyi, esprima does it the same way.

**Comment by nzakas** [comment]
That doesn't mean it's correct ;)

The intent is to see how possible or is to always return the first ast even without semicolons.

**Comment by nzakas** [comment]
Fixed with 3.0.0.
